modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/idp/mgt/IdentityProviderMgtServiceTestCase.java

```diff
@@ -360,7 +370,187 @@ public void testGetAllIdPs() throws Exception {
+        }
+    }
+
+    @Test(priority = 5, groups = "wso2.is", description = "test getAllPaginatedIdPsInfo operation with invalid page number")
+    public void testGetPaginatedIdPsWithInvalidPageNumber() {
+
+        try {
+            int pageNumber = -1;
+            idpMgtServiceClient.getAllPaginatedIdPsInfo(pageNumber);
+        } catch (Exception e) {
+            Assert.assertEquals(e.getMessage(), invalidPageNumberErrorMessage, "Error message of invoking " +
+                    "getAllPaginatedIdPsInfo with invalid page number did not match with the expected error message: " +
+                    invalidPageNumberErrorMessage);
+        }
+    }
+
+    @Test(priority = 6, groups = "wso2.is", description = "test getAllPaginatedIdPsInfo operation with wrong page number")
+    public void testGetPaginatedIdPsWithIncorrectPageNumber() throws Exception {
+
+        int pageNumber = 2;
+        List<IdentityProvider> idpList = idpMgtServiceClient.getAllPaginatedIdPsInfo(pageNumber);
+        if (idpList.size() > 0) {
+            Assert.fail("List of Idps found while calling getAllPaginatedIdPsInfo() with wrong page number.");
+        }
+    }
+
+    @Test(priority = 7, groups = "wso2.is", description = "test getPaginatedIdPsInfo operation")
+    public void testGetAllPaginatedIdPsWithFilter() throws Exception {
+
+        String filter = "name sw \"Search\" and name ew \"Test\"";
+        int pageNumber = 1;
+        List<IdentityProvider> idpList = idpMgtServiceClient.getPaginatedIdPsInfo(filter, pageNumber);
+        Assert.assertNotNull(idpList);
+        if (idpList.size() > 0) {
+            IdentityProvider filteredIdp = null;
+            for (IdentityProvider idp : idpList) {
+                if (testIdpNameSearch.equals(idp.getIdentityProviderName())) {
+                    filteredIdp = idp;
+                }
+            }
+            Assert.assertNotNull(filteredIdp, "Searched Idp not found");
+        } else {
+            Assert.fail("Unable to find filtered identity provider. Paginated IDP list is empty.");
+        }
+    }
+
+    @Test(priority = 8, groups = "wso2.is", description = "test getPaginatedIdPsInfo operation")
+    public void testGetAllPaginatedIdPsWithEmptyFilter() throws Exception {
+
+        String filter = "";
+        int pageNumber = 1;
+        List<IdentityProvider> idpList = idpMgtServiceClient.getPaginatedIdPsInfo(filter, pageNumber);
+        Assert.assertNotNull(idpList);
+        if (idpList.size() > 0) {
+            int filteredIdpSize = 0;
+            for (IdentityProvider idp : idpList) {
+                if (testIdpName.equals(idp.getIdentityProviderName()) ||
+                        testIdpNameSearch.equals(idp.getIdentityProviderName())) {
+                    filteredIdpSize++;
+                }
+            }
+            Assert.assertEquals(filteredIdpSize, 2, "filtered identity provider size not matched" +
```

Comment:
why we need to get the filteredIdpSize here can't we use 
Assert.assertEquals(idpList.size(), 2) here?